types/index.d.ts

```diff
@@ -836,6 +836,25 @@ declare module 'cloudinary' {
         value: string;
     }
 
+    export type AnalysisType = 'captioning' | 'cld_fashion' | 'cld_text' | 'coco' | 'google_tagging' | 'human_anatomy' | 'lvis' | 'shop_classifier' | 'unidet' | 'custom';
+
+    export interface AnalyzeResponse {
+        data: {
+            entity: string,
+            analysis: Record<string, Record<string, string> | Array<string> | string>
```

Comment:
Is this generally how custom JSON is represented?

Reply:
Yes, `Record<>` is object, later on you specify the type for keys and values. So in this case `analysis` can be an object with strings as keys and another object as value or an array with strings or just a string.